lib/mayaUsd/fileio/translators/translatorMayaReference.cpp

```diff
@@ -424,29 +472,74 @@ MStatus UsdMayaTranslatorMayaReference::update(const UsdPrim& prim, MObject pare
     // to be a common thing in the workflow.)
     //
     if (refNode.isNull()) {
-        for (MItDependencyNodes refIter(MFn::kReference); !refIter.isDone(); refIter.next()) {
-            MObject      tempRefNode = refIter.item();
-            MFnReference tempRefFn(tempRefNode);
-            if (!tempRefFn.isFromReferencedFile()) {
-
-                // Get the name of the reference node so we can check if this node
-                // matches the prim we are processing.  Strip off the "_RN" suffix.
-                //
-                MString refNodeName = tempRefFn.name();
-                MString refName(refNodeName.asChar(), refNodeName.numChars() - 3);
-
-                // What reference node name is the prim expecting?
-                MString expectedRefName = refNameFromPath(parentDag);
-
-                // If found a match, reconnect the reference node's `associatedNode`
-                // attr before loading it, since the previous connection may be gone.
-                //
-                if (refName == expectedRefName) {
-                    // Reconnect the reference node's `associatedNode` attr before
-                    // loading it, since the previous connection may be gone.
-                    connectReferenceAssociatedNode(parentDag, tempRefFn);
-                    refNode = tempRefNode;
-                    break;
+        if (isMayaReference(prim)) {
+            // New behaviour for MayaReference
```

Comment:
The one thing I worry about with the session layer storage, is how smooth the experience is for a user that wants to "reconnect" a reference. If a user wants to swap a reference in and connect it to a stage, is this how you see it working?

- existing ref (ref01_RN)
- bring in reference (named ref02_RN)
- set session layer data to point to ref02_RN (maybe via a tool)
- refresh prim... new ref is connected... and name is updated to ref01_RN but fails (hopefully it doesn't become ref03_RN)

Reply:
This looks like the case in your test from https://github.com/Autodesk/maya-usd/pull/1620, let see how it behaves.

Reply:
I do think this test would be useful but we can tackle it in a follow up MR if you wish.